### ISSUE TITLE
Group S Challenge

### ISSUE THREAD

**Issue by Jorerim** [Cxbx-Reloaded/game-compatibility#796]
Description
Goes ingame, till the start of the race and then it just stop the emulation. The buttons "A" and "B" are pressed by itself, need to press "A" very quick to go through the menu otherwise it goes a menu back.
Using HLE, no GPU LLE.

Dumps
[Xbe.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/2087399/Xbe.txt)
[KrnlDebug.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/2087400/KrnlDebug.txt)
![untitled](https://user-images.githubusercontent.com/38312637/41196650-ebee1d78-6c1b-11e8-92a8-56f72c945ce6.png)
![untitled2](https://user-images.githubusercontent.com/38312637/41196651-ef3cf2d8-6c1b-11e8-9148-61ec4dc3f095.png)
![untitled3](https://user-images.githubusercontent.com/38312637/41196652-f2294564-6c1b-11e8-9963-34cb3e60b706.png)

**Comment by Jorerim** [comment]
Fixed xbe dump should be unmodified now

**Comment by SonicThePS4** [comment]
Yup, the button is fixed. But as soon when the camera got close to the Dodge Viper GTS to change the racing camera view, it gave us the Fatal Error. But Group S Challenge is making it's progress pretty well.

![image](https://user-images.githubusercontent.com/39503785/60117267-ba904500-9747-11e9-94f7-5be975154473.png)

**Comment by CookiePLMonster** [comment]
Can this be re-tested? A 1.05 version of the same game instantly crashes on startup, so it would be good to know if 1.03 also broke (regression) or if it still works fine (version difference).

**Comment by ObiKKa** [comment]
This [PR # 2048](https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/pull/2048#issue-528235028) (Optimize vertex stream cache for partial buffer access), dated November 27, 2020, stated and showed that this game was supposed to perform a lot faster than merely 1-2 FPS on average.

**PR's note:** Improves in-race performance in Group S Challenge by over 10 times (3 FPS -> ~40 FPS), and makes the emulator not run out of memory after a few seconds into the race. The same is still too slow and buggy to be playable.

**Comment by rozniak** [comment]
Have tested on `811f173`.

Game seems to be largely fine for me, movies and menus work as expected. Gameplay works however there are hall-of-mirrors effects throughout on the road and the sky.

Performance is decent, I get a reliable 60FPS when in test drive mode (just my car), it does dip to the high 20s when back of the pack during an actual race however might want someone with a beefier CPU to test.

My specs:
CPU - AMD FX 6350 @ 3.9GHz
GPU - NVIDIA GTX 1070 3GB
RAM - 16GB DDR3

Screenies:
![cxbx_rdOllB7xv9](https://user-images.githubusercontent.com/13258281/117195977-d9e56800-addd-11eb-8c14-f37389471c5a.png)
![cxbx_aZEu8nlrNK](https://user-images.githubusercontent.com/13258281/117195990-dfdb4900-addd-11eb-8e2c-9a8e8d8656e3.png)
![cxbx_1bhODtL6Y1](https://user-images.githubusercontent.com/13258281/117196035-e9fd4780-addd-11eb-8217-53f879893a61.png)
![cxbx_DhkPW4bS7D](https://user-images.githubusercontent.com/13258281/117196050-ee296500-addd-11eb-9352-99b125e0f9b6.png)
![cxbx_vob35rpVE2](https://user-images.githubusercontent.com/13258281/117196060-f08bbf00-addd-11eb-8e9c-2ae0d45a9df3.png)

[KrnlDebug.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/6429674/KrnlDebug.txt)
